convert/harness/downgrader/downgrade.go

```diff
@@ -532,3 +540,98 @@ func convertImagePull(v string) (s string) {
 		return ""
 	}
 }
+
+func convertWhen(when *v1.When, stepId string) *v0.When {
+	if when == nil {
+		return nil
+	}
+
+	newWhen := &v0.When{
+		StageStatus: "Success", // default
+	}
+	var conditions []string
+
+	for _, cond := range when.Cond {
+		for k, v := range cond {
+			switch k {
+			case "event":
+				if v.In != nil {
+					var eventConditions []string
+					for _, event := range v.In {
+						eventConditions = append(eventConditions, fmt.Sprintf("<+trigger.event> =~ %q", event))
+					}
+					conditions = append(conditions, fmt.Sprintf("%s", strings.Join(eventConditions, " || ")))
+				}
+				if v.Not != nil && v.Not.In != nil {
+					var notEventConditions []string
+					for _, event := range v.Not.In {
+						notEventConditions = append(notEventConditions, fmt.Sprintf("<+trigger.event> !~ %q", event))
+					}
+					conditions = append(conditions, fmt.Sprintf("%s", strings.Join(notEventConditions, " && ")))
+				}
+			case "status":
+				if v.Eq != "" {
+					newWhen.StageStatus = v.Eq
+				}
+				if v.In != nil {
+					var statusConditions []string
+					for _, status := range v.In {
+						statusConditions = append(statusConditions, fmt.Sprintf("<+execution.steps.%s.status> =~ %q", stepId, status))
+					}
+					conditions = append(conditions, fmt.Sprintf("%s", strings.Join(statusConditions, " || ")))
+				}
+			case "branch":
+				if v.In != nil {
+					var branchConditions []string
+					for _, branch := range v.In {
+						branchConditions = append(branchConditions, fmt.Sprintf("<+trigger.branch> =~ %q", branch))
+					}
+					conditions = append(conditions, fmt.Sprintf("%s", strings.Join(branchConditions, " || ")))
+				}
+				if v.Not != nil && v.Not.In != nil {
+					var notBranchConditions []string
+					for _, branch := range v.Not.In {
+						notBranchConditions = append(notBranchConditions, fmt.Sprintf("<+trigger.branch> !~ %q", branch))
+					}
+					conditions = append(conditions, fmt.Sprintf("%s", strings.Join(notBranchConditions, " && ")))
+				}
+			case "repo":
+				if v.In != nil {
+					var repoConditions []string
+					for _, repo := range v.In {
+						repoConditions = append(repoConditions, fmt.Sprintf("<+trigger.payload.repository.name> =~ %q", repo))
+					}
+					conditions = append(conditions, fmt.Sprintf("%s", strings.Join(repoConditions, " || ")))
+				}
+				if v.Not != nil && v.Not.In != nil {
+					var notRepoConditions []string
+					for _, repo := range v.Not.In {
+						notRepoConditions = append(notRepoConditions, fmt.Sprintf("<+trigger.payload.repository.name> !~ %q", repo))
+					}
+					conditions = append(conditions, fmt.Sprintf("%s", strings.Join(notRepoConditions, " && ")))
+				}
+			case "ref":
+				if v.In != nil {
+					var refConditions []string
+					for _, ref := range v.In {
+						refConditions = append(refConditions, fmt.Sprintf("<+codebase.commitRef> =~ %q", ref))
```

Comment:
We probably can't use `<+codebase.commitRef>` since it will not be available for trigger events.

`<+trigger.payload.ref>` is an alternative, but it likely differs between SCMs.